pkg/api/admin/openshiftcluster.go

```diff
@@ -441,6 +441,7 @@ type UserAssignedIdentities map[string]ClusterUserAssignedIdentity
 type Identity struct {
 	Type                   string                 `json:"type,omitempty"`
 	UserAssignedIdentities UserAssignedIdentities `json:"userAssignedIdentities,omitempty"`
+	IdentityURL            string                 `json:"identityURL,omitempty"`
```

Comment:
Would it ever be useful in a Geneva Action context? I think that would be the use case for keeping it admin API, but I'm not sure of others.

Reply:
To clarify:
- We need to persist to fetch the credentials object
- After we fetch the credentials object, we store in key vault. That object has the identity URL (field ClientSecretURL): https://github.com/Azure/msi-dataplane/blob/557894f3863f5591445903674cc6ba90e2a26fb7/pkg/dataplane/zz_generated_models.go#L21
- Credential refresher will use the URL stored in key vault for fetching future credentials objects.

There are a few gotchas for line 445:
1. MSI team reserves the right to change the identity URL after refresh, so if we persist this guy, we should at least have a code comment clarifying when and when not to use this field. https://eng.ms/docs/products/arm/rbac/managed_identities/msicredentialshandlingguideline
2. Instead of relying on Geneva Action to see if this value got persisted, couldn't we just rely on log statements: (1) did we get a valid identity URL and (2) did we persist it

Reply:
Yes, it's better to rely on the Service Logs for this purpose, and this will only be faced for the Cluster Installs. Since, this field would be anyways removed after the bootstrap MSI certificate fetching.

Reply:
I'll remove this from the admin API for now.